File: week6/sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

# Set baseline values for counting number of letters, words, and sentences in text
letters = 0
words = 1
sentences = 0

# Get number of letters, words, and sentences
for char in text:

    if char.isalpha():
        letters += 1

    elif char.isspace():
        words += 1

    elif char == "." or char == "!" or char == "?":
        sentences += 1

# Avg number of letters per 100 words, L
L = (letters / words) * 100

# Average number of sentences per 100 words, S
S = (sentences / words) * 100

# Calculate Coleman-Liau index
index = 0.0588 * L - 0.296 * S - 15.8

# Check if index is lower than 1 or above 16, and respond as indicated in instructions
# If not, print the grade level
if index < 1:
    print("Before Grade 1")

elif index >= 16:
    print("Grade 16+")

else:
    print(f"Grade {round(index)}")

File: week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

// prototype functions
int count_letters(string text);

int count_words(string text);

int count_sentences(string text);

int main(void)
{
    // Prompt user for text
    string text = get_string("Text: ");
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);
    int index = round((0.0588 * (letters / words) * 100 - 0.296 * (sentences / words) * 100) - 15.8);
    // print grade level of text
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

// Function to count letters in text
int count_letters(string text)
{
    int letters = 0;
    int length = strlen(text);
    // Get number of letters in string
    for (int i = 0; i < length; i++)
    {
        if (isupper(text[i]) || islower(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1;
    int length = strlen(text);
    //Get number of words in a string
    for(int i = 0; i < length; i++)
    {
        if ((text[i]) == ' ')
        {
            words++;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    int length = strlen(text);
    for (int i = 0; i < length; i++)
    {
        if ((text[i]) == '.' || (text[i]) == '!' || (text[i]) == '?')
        {
            sentences++;
        }
    }

    return sentences;
}